### ISSUE TITLE
Canary Deployment patterns (spike)

### ISSUE THREAD

**Issue by evanlouie** [CatalystCode/kubemalt#40]
As an operator, I want best practices on how to do A/B testing and canary testing within the context of Istio.

**Comment by evanlouie** [comment]
https://dev.azure.com/indiasfdevcrew/SamsCXOSprints/_workitems/edit/2

**Comment by bnookala** [comment]
Question raised by @dtzar: will istioctl be a hard requirement for applying istio virtualservice rule updates?

**Comment by andrebriggs** [comment]
> Also to investigate: azure devops plugin for gated rollouts based on monitoring and metrics gathered from prometheus/grafana

The above sounds like a separate issue. We have various meeting notes on the strategy able being alluded to.

> @andrebriggs - This will impact how your team deploys and configures flux/azure devops for multiple clusters.

Agreed. This scenario is a "run" stage scenario as opposed to the current "crawl" and "walk" scenarios

**Comment by evanlouie** [comment]
No requirement specifies the use of `istioctl`. In fact, if possible, I would stay away from `istioctl`. I think it should be manageable for most operators to manage even complex routing rules with manifests alone.

**Comment by dtzar** [comment]
yes @andrebriggs I agree it should be a separate task.  I created this one: https://github.com/CatalystCode/kubemalt/issues/45
